contracts/test/MockUpgradedBondFactory.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "../bond/CollateralWhitelist.sol";
+import "../bond/ERC20SingleCollateralBond.sol";
+import "../bond/BondCreator.sol";
+import "../bond/Roles.sol";
+import "../Version.sol";
+
+contract UpgradedVersion {
+    string public constant VERSION = "mock_tag";
+}
+
+contract MockUpgradedBondFactory is
```

Comment:
this contract was needed for testing the VERSION proxy upgradability I think? Could stub out/remove some of the functionality?

Reply:
The `windranger-solidity-template` has contracts demoing upgradability with tests 
 e.g. https://github.com/windranger-io/windranger-solidity-template/blob/main/contracts/test/upgradable/BoxExtension.sol
 https://github.com/windranger-io/windranger-solidity-template/blob/main/test/box-upgradability.test.ts
 
 The contract is pretty big, I'd favour following the templates approach (as that's where the version contract will get promoted too), with a simple upgradable box contract to demo the version changing wdyt?

Reply:
Ok I think I have this done.